pkg/lint/fileparser.go

```diff
@@ -0,0 +1,287 @@
+	tf := &TemplateFile{
+		filepath:    filepath,
+		sourceBytes: body,
+	}
+
+	// extract templated values and all paths from the template
+	values := map[string]*TemplateValue{}
+	paths := map[string]bool{}
+	{
+		includes.clear()
+		t, err := template.
+			New(filepath).
+			Funcs(fMap).
+			Option("missingkey=zero").
+			Parse(string(body))
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+		tf.sourceTemplate = t
+
+		// extract all values
+		for _, node := range t.Tree.Root.Nodes {
+			if node.Type() == parse.NodeText {
+				continue
+			}
+
+			nodePaths := templatePathPattern.FindAllString(node.String(), -1)
+			for _, np := range nodePaths {
+				normalPath := NormalPath(np)
+				if _, ok := values[normalPath]; !ok {
+					values[normalPath] = &TemplateValue{
+						Path:            normalPath,
+						OccurrenceCount: 1,
+					}
+				} else {
+					values[normalPath].OccurrenceCount += 1
+				}
+			}
+		}
+
+		// extract all paths
+		output := bytes.NewBuffer([]byte{})
+		var data interface{}
+		// Render template without values. All templated values will be
+		// replaced by default zero values: "" for string, 0 for int, false
+		// for bool etc.
+		err = t.Execute(output, data)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+		tf.includes = includes.Filepaths
+
+		c := pathmodifier.Config{
+			InputBytes: output.Bytes(),
+			Separator:  ".",
+		}
+
+		svc, err := pathmodifier.New(c)
+		if err != nil {
+			// try to pretty print offending yaml
+			fmt.Println("debug ---")
+			fmt.Println(output.String())
+			fmt.Println("debug ---")
+			var yamlOut interface{}
+			yamlErr := yaml.Unmarshal(output.Bytes(), &yamlOut)
+
+			if yamlErr == nil {
+				return nil, microerror.Mask(err)
+			}
+
+			matches := yamlErrorLinePattern.FindAllStringSubmatch(yamlErr.Error(), -1)
+			if len(matches) == 0 {
+				return nil, microerror.Mask(err)
+			}
+
+			lineNo, convErr := strconv.Atoi(matches[0][1])
+			if convErr != nil {
+				return nil, microerror.Mask(err)
+			}
+
+			lines := strings.Split(output.String(), "\n")
+			fmt.Println(red(yamlErr.Error()))
+			fmt.Println("In " + filepath)
+			if lineNo > 1 {
+				fmt.Println("> " + lines[lineNo-2])
+			}
+			fmt.Println("> " + red(lines[lineNo-1]))
+			if lineNo < len(lines)-2 {
+				fmt.Println("> " + lines[lineNo])
+			}
+
+			return nil, microerror.Mask(err)
+		}
+
+		pathList, err := svc.All()
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		for _, p := range pathList {
+			paths[p] = true
+		}
+	}
+	tf.values = values
+	tf.paths = paths
+
+	// fill in installation and app if possible
+	{
+		elements := strings.Split(filepath, "/")
+		if strings.HasPrefix(filepath, "installations") {
+			tf.installation = elements[1]
+			tf.app = elements[3]
+		} else if strings.HasPrefix(filepath, "default") {
+			tf.app = elements[2]
+		}
+		// else it's an include file and has neither app nor installation
+	}
+
+	return tf, nil
+}
+
+func NormalPath(path string) string {
```

Comment:
We can simply replace this with `strings.TrimPrefix` - it checks if the string has the prefix already. And even if it wasn't performance gain here is probably neglectable.

Reply:
Done.